### DISCUSSION TITLE
Caching of responses in the front and/or backend

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [trpc/trpc#4831] [Q&A]
How can I cache my responses in the frontend and/or the backend with tRPC with Next.js and the pages directory?

**Comment by FleetAdmiralJakob** [comment]
Backend case: So, for example, I can say that if someone makes the exact same request in the same hour he gets the cached data.

Frontend case: I'm realoading the page or come back to the page in the same hour and now request is made because it's frontend-side cached

Is at least one of this cases possible with tRPC?

**Reply by akparhi** [reply]
If you need to: 
- use memoize in your backend (Something like https://www.npmjs.com/package/memoize-one)
- For FE https://tanstack.com/query/v4/docs/react/plugins/persistQueryClient

**Reply by FleetAdmiralJakob** [reply]
Hey @akparhi, thank you so much for your answer! Do you know how this memoize thing works? Like how is it storing the stuff and how can I even use it with tRPC?

**Reply by FleetAdmiralJakob** [reply]
I created another discussion for this: https://github.com/trpc/trpc/discussions/4909
